### ISSUE TITLE
DKIM-signing is breaking MIME format

### ISSUE THREAD

**Issue by mpizzoli** [vstakhov/rspamd#1722]
### Classification (Please choose *one* option):

* [ ] Crash/Hang/Data loss
* [ ] WebUI/Usability
* [x] Serious bug
* [ ] Other bug
* [ ] Feature
* [ ] Enhancement

### Reproducibility (Please choose *one* option):

* [X] Always
* [ ] Sometimes
* [ ] Rarely
* [ ] Unable
* [ ] I didn’t try
* [ ] Not applicable

### Rspamd version:
1.6.1

### Operation system, CPU, memory and environment:
RHEL7 x86_64

### Description (Please provide a descriptive summary of the issue):
Applying the DKIM signature is creating non-MIME-comformant emails.

I don't see issues in sending i.e. to gmail and yahoo, but I am seeing it sending to a domain protected by a (picky) Symantec Brightmail appliance which is reporting that the MIME format is incorrect.

I checked the very same email, side by side with the correspondent non-DKIM-signed version.
The dkim-signed one got anomalous scores in the receiving system.

I compared the DKIM header of the RSPAMD email to another signed email. Here what I see:

Grouponmail: 
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=r.grouponmail.it; s=s2048d20160829; t=1499091027; bh=eekgh06Fd73qz95ph20GWhsVmIr7Op3KQr0yexkV1PA=; h=Date:From:To:Message-ID:Subject:Content-Type; b=hl1Gt+jpP59+l6Fi0L8amB9Xjt75b+Pbyw+j8tIXbvl49ZpkwHSlyoN9D+7rsapcs
	 jJCKi02rQWsB7KM326PbfJjN308AEGeHWbupWcxawpvUuDKiAF4LfRIz0I7WmkNuxH
	 Pg17q1mfooVOTEvdiLSZIRZgc3wEzSS+L7Z8mVEU6nVE+nCp8Ng8qhoL6bX0anhoQX
	 zNlVZbALIXrY8G++1l0d1AjMZf8NTfpR/Jv4V0s2uFomuZEvQRS5nMtWnaTMbrZRJl
	 yj2r/vAZlFQfB9mwr/y7oycaNSMqyqfSmgioDM71wNIE44QLSSo+JYtRmofnGRtySH
	 B0qKJtNASNFYQ==

Rspamd signed header:
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=newsletter.cornercard.ch; s=key-20170619-2048-bim02; t=1499089945; h=from:sender:reply-to:subject:date:message-id:to:mime-version:content-type:list-id:list-unsubscribe; bh=5VYp0+eQWIM/AUfqzLqoqnFlXK0v6djsNyVMYyrV2Yk=; b=wPF9LS61/Dipif7dY3hx2AssYLNBqgItOAmw9v1aau4ITtKVbwIPjWZn9aKvVCED8xGgfB RH2xmJFYWSAHElLYseveUI6oTGDQhOaaEmakffBwKcm7ISWqYzBCpqaLZrUnoPfESt2Ivf ZdQDDVYwhh6T5+LjGW4udfEUO5EsSVfnZT2JNyZkfU/msX0ZYe5g0/Q6nR8L9kRvVFWquq in3bZ0VoHazF/UziWEfeEPkaBvDEG8yD16/JAyF/LgvZwwyq1sFBWhZt6owPqg/gt55EGf WIsNIlmUC1ZGFQbYwWjthNJZiy22HULrp78v+IG491ueDVIRSfxV6My8XFAsKQ==

As you can see, rspamd doesn't honor the indenting, making it appear as multiple (wrong) headers.

In case of need, I can privately send the rspamd-signed email. 

### Compile errors (if any):
N/A

### Steps to reproduce:
Just send email via dkim_signing module

### Expected results:
MIME compliant email

### Actual results:
Wrong MIME format

### Debugging information (see details [here](https://rspamd.com/doc/faq.html#how-to-figure-out-why-rspamd-process-crashed)):
N/A

### Configuration:
Non important (?)

### Additional information:
N/A

**Comment by mpizzoli** [comment]
Github web interface removed the whitespaces.
I try to correct the report here down.


Grouponmail:

DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=r.grouponmail.it; s=s2048d20160829; t=1499091027; bh=eekgh06Fd73qz95ph20GWhsVmIr7Op3KQr0yexkV1PA=; h=Date:From:To:Message-ID:Subject:Content-Type; b=hl1Gt+jpP59+l6Fi0L8amB9Xjt75b+Pbyw+j8tIXbvl49ZpkwHSlyoN9D+7rsapcs
<tab><space>jJCKi02rQWsB7KM326PbfJjN308AEGeHWbupWcxawpvUuDKiAF4LfRIz0I7WmkNuxH
<tab><space>Pg17q1mfooVOTEvdiLSZIRZgc3wEzSS+L7Z8mVEU6nVE+nCp8Ng8qhoL6bX0anhoQX
<tab><space>zNlVZbALIXrY8G++1l0d1AjMZf8NTfpR/Jv4V0s2uFomuZEvQRS5nMtWnaTMbrZRJl
<tab><space>yj2r/vAZlFQfB9mwr/y7oycaNSMqyqfSmgioDM71wNIE44QLSSo+JYtRmofnGRtySH
<tab><space>B0qKJtNASNFYQ==


Rspamd:
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=newsletter.cornercard.ch; s=key-20170619-2048-bim02; t=1499089945; h=from:sender:reply-to:subject:date:message-id:to:mime-version:content-type:list-id:list-unsubscribe; bh=5VYp0+eQWIM/AUfqzLqoqnFlXK0v6djsNyVMYyrV2Yk=; b=wPF9LS61/Dipif7dY3hx2AssYLNBqgItOAmw9v1aau4ITtKVbwIPjWZn9aKvVCED8xGgfB RH2xmJFYWSAHElLYseveUI6oTGDQhOaaEmakffBwKcm7ISWqYzBCpqaLZrUnoPfESt2Ivf ZdQDDVYwhh6T5+LjGW4udfEUO5EsSVfnZT2JNyZkfU/msX0ZYe5g0/Q6nR8L9kRvVFWquq in3bZ0VoHazF/UziWEfeEPkaBvDEG8yD16/JAyF/LgvZwwyq1sFBWhZt6owPqg/gt55EGf WIsNIlmUC1ZGFQbYwWjthNJZiy22HULrp78v+IG491ueDVIRSfxV6My8XFAsKQ==

**Comment by mpizzoli** [comment]
I fail to get the desired behaviour, and from my company I cannot reach pastebin, sorry.

**Comment by moisseev** [comment]
@mpizzoli , You can use markdown formatting for code blocks https://help.github.com/articles/creating-and-highlighting-code-blocks/ or attach files to the isssue.

**Comment by mpizzoli** [comment]
Got it! @moisseev  thanks!

Grouponmail:

```
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=r.grouponmail.it; s=s2048d20160829; t=1499091027; bh=eekgh06Fd73qz95ph20GWhsVmIr7Op3KQr0yexkV1PA=; h=Date:From:To:Message-ID:Subject:Content-Type; b=hl1Gt+jpP59+l6Fi0L8amB9Xjt75b+Pbyw+j8tIXbvl49ZpkwHSlyoN9D+7rsapcs
	 jJCKi02rQWsB7KM326PbfJjN308AEGeHWbupWcxawpvUuDKiAF4LfRIz0I7WmkNuxH
	 Pg17q1mfooVOTEvdiLSZIRZgc3wEzSS+L7Z8mVEU6nVE+nCp8Ng8qhoL6bX0anhoQX
	 zNlVZbALIXrY8G++1l0d1AjMZf8NTfpR/Jv4V0s2uFomuZEvQRS5nMtWnaTMbrZRJl
	 yj2r/vAZlFQfB9mwr/y7oycaNSMqyqfSmgioDM71wNIE44QLSSo+JYtRmofnGRtySH
	 B0qKJtNASNFYQ==
```
Rspamd:
```
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=newsletter.cornercard.ch; s=key-20170619-2048-bim02; t=1499089945; h=from:sender:reply-to:subject:date:message-id:to:mime-version:content-type:list-id:list-unsubscribe; bh=5VYp0+eQWIM/AUfqzLqoqnFlXK0v6djsNyVMYyrV2Yk=; b=wPF9LS61/Dipif7dY3hx2AssYLNBqgItOAmw9v1aau4ITtKVbwIPjWZn9aKvVCED8xGgfB RH2xmJFYWSAHElLYseveUI6oTGDQhOaaEmakffBwKcm7ISWqYzBCpqaLZrUnoPfESt2Ivf ZdQDDVYwhh6T5+LjGW4udfEUO5EsSVfnZT2JNyZkfU/msX0ZYe5g0/Q6nR8L9kRvVFWquq in3bZ0VoHazF/UziWEfeEPkaBvDEG8yD16/JAyF/LgvZwwyq1sFBWhZt6owPqg/gt55EGf WIsNIlmUC1ZGFQbYwWjthNJZiy22HULrp78v+IG491ueDVIRSfxV6My8XFAsKQ==`
```

**Comment by vstakhov** [comment]
It is known issue for 1.6.1. It is fixed in master branch and experimental packages.

**Comment by andryyy** [comment]
It will be included in 1.6.2 he said on IRC. :-)

**Comment by mpizzoli** [comment]
Thank you very much for the info. Closing the ticket
